connectors/src/connectors/slack/index.ts

```diff
@@ -617,9 +622,32 @@ export class SlackConnectorManager extends BaseConnectorManager<SlackConfigurati
   }: {
     internalId: string;
   }): Promise<Result<string[], Error>> {
-    // We only ever return permissions at the slack channel level so the parents are always the
-    // internalId itself.
-    return new Ok([internalId]);
+    // If the internal ID is a Slack channel ID, it has no other parent
+    if (isSlackChannelInternalId(internalId)) {
+      return new Ok([internalId]);
+    }
+    // If it is a slack thread, or a slack "non-threaded" message document, it also
+    // needs the channel internal ID as parent
+    else if (isSlackThreadInternalId(internalId)) {
+      const { channelId } =
+        slackThreadIdentifierFromSlackThreadInternalId(internalId);
+      return new Ok([
+        internalId,
+        slackChannelInternalIdFromSlackChannelId(channelId),
+      ]);
+    } else if (isSlackNonThreadedMessagesInternalId(internalId)) {
+      const channelId =
+        slackChannelIdFromSlackNonThreadedMessagesInternalId(internalId);
+      return new Ok([
+        internalId,
+        slackChannelInternalIdFromSlackChannelId(channelId),
+      ]);
+    }
+    // This in theory shouldn't happen
+    else {
+      logger.warn({ internalId }, "Unknown internal ID for Slack connector");
```

Comment:
would be stronger here, like at first panic log
if this happens we should fix

Reply:
aligned